### DISCUSSION TITLE
Waveshare 2 or 4 Port RS485 Converter

### DISCUSSION THREAD

**Discussion by MrJoki007** [alienatedsec/solis-ha-modbus-cloud#29] [Q&A]
This converter should work two right?
https://www.amazon.de/Waveshare-2-CH-RS485-Converter-Independent/dp/B0CC25WLNJ/ref=sr_1_3?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=1CWZTL3FBWQ2M&keywords=2-Ch+RS485+to+RJ45+Ethernet&qid=1705741513&sprefix=2-ch+rs485+to+rj45+ethernet+%2Caps%2C481&sr=8-3

It has two RS485 ports and a build in ethernet switch. So it would suit the application perfectly.

Anotherone could be the four port one, if you want to monitor your battery two.

https://www.amazon.de/Waveshare-Independent-Bi-Directional-Transparent-Transmission/dp/B0CQYGWBMZ/ref=sr_1_4?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=32XYG5PNQ99WW&keywords=waveshare+rs485+4ch&qid=1705737302&sprefix=waveshare+rs485+4+ch%2Caps%2C107&sr=8-4

**Comment by alienatedsec** [comment]
It's unlikely to work the same way. The server-client logic between RS485 master and slave would need to be managed on the same device, which I don't think is possible. 

Did you buy it? If so, can you share any screenshots of the config webpage?

**Reply by MrJoki007** [reply]
I bought the one with two network and two rs485 ports.
It will be delivered on thuesday

**Reply by alienatedsec** [reply]
@MrJoki007 If the below is correct, the device will not allow comms between RS485 connections.

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/e225ad2a-0522-46ed-a98c-d95f004c65b0)

It uses the same VirCom software, which sets the device mode to `TCP Server` or `TCP Client`; you cannot set both modes to communicate between two RS485 ports on the same device.

**Reply by alienatedsec** [reply]
In short, you need your device connected to confirm fully; however........

As always, Waveshare's documentation is vague and mentions something but does not show it correctly.

If you can find two interfaces in the following window, it could mean that one is used for the `RS485_1` and the other is for the `RS485_2`
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/eac45e4e-dd1a-48ed-b67b-35b9494582a7)

It shows communication between two interfaces here:
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/2c33bc10-7900-42a7-849f-fdb2f55c376e)

**Reply by MrJoki007** [reply]
![grafik](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/8217564/a832592f-4f98-4d0e-99af-2c614868a283)
So it should work right?

**Reply by alienatedsec** [reply]
@MrJoki007 wow - yes that should work :) As much as I was sceptical, it works the way I read it :)

So, it should work if you could follow the same configuration as in my repo for `TCP Client` and `TCP Server`.

**Reply by MrJoki007** [reply]
It works like a charm.
And like this there’s no switch required
I just have to tidy it up a bit

![IMG_1632](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/8217564/886948dc-ad9b-4892-9a1c-cc6eee10f844)

![IMG_1633](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/8217564/595a3092-5562-483c-bc64-a0c4839894c4)

**Reply by alienatedsec** [reply]
Are you using the POE HAT powered by the Inverter, and that's to power the Waveshare in one Ethernet port and the other Ethernet is used for comms?

**Reply by MrJoki007** [reply]
No, the waveshare bridge is powered by an external power supply.
The DLS-L is powered by the inverter.
Path rs485: inverter -> waveshare rs485-1 -> network -> waveshare rs485-2 -> DLS-L
Path network: Switch -> waveshare -> DLS-L

**Reply by alienatedsec** [reply]
@MrJoki007  I just noticed you disassembled your `DLS-L` :) 

Is your `DLS-L` not working with Home Assistant? it should have the Modbus TCP support

**Reply by MrJoki007** [reply]
No that doesn’t work because my serial number starts with 1913 and not 1920

**Reply by alienatedsec** [reply]
@MrJoki007 
I guess it could be also firmware-related - if you wish, I have several versions available, but if your solution is working now, then there may not be a need to do so.

**Reply by MrJoki007** [reply]
My current firware on the stick is ME-1-2209070U
ME-121001-V1.0.6(202209071030)
What firmware do you have?
![SOLIS Datenlogger Matrix](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/8217564/3a6caa91-0742-426e-85fb-f2a681c60f82)

**Reply by alienatedsec** [reply]
@MrJoki007 

It looks like the latest firmware - is connecting to port 3389 not working?

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/544d69f7-b229-4ed4-8a9c-dc45782371ea)

**Reply by MrJoki007** [reply]
![grafik](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/8217564/57c27b50-e700-4829-b093-fc15e083212c)
Solis is over the waveshare adapter
test is, when i try to connect directly to the DLS-L

**Reply by alienatedsec** [reply]
@MrJoki007 Unlucky

**Reply by MrJoki007** [reply]
Yes, but i think the solution with the dual wireshark module is very cool and it works well

**Reply by SmileyMan** [reply]
![PXL_20240917_161158079](https://github.com/user-attachments/assets/5d41785e-498b-40cc-b54b-a1281300252a)

Confirm this works.
